server/index.js

```diff
@@ -481,43 +483,64 @@ function updateSessionState( session, setState=null ) {
   }
 }
 
-function resetSession( session ) {
+function resetSession(session) {
   // Reset Time
-  resetSessionTime( session );
+  resetSessionTime(session);
 
   // Reset Positions & Angles
-  Object.keys( serverData[ "sessions" ][ session ][ "players" ] ).forEach( ( player ) => {
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "spaceship" ][ "position" ] = [ 50, 50 ];
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "spaceship" ][ "angle" ] = 0;
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "spaceship" ][ "resources" ] = 0;
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "planet" ][ "position" ] = [ 50, 50 ];
-    serverData[ "sessions" ][ session ][ "players" ][ player ][ "planet" ][ "resources" ] = 0;
-  } );
+  Object.keys(serverData["sessions"][session]["players"]).forEach((player) => {
+    serverData["sessions"][session]["players"][player]["spaceship"]["position"] = [50, 50];
+    serverData["sessions"][session]["players"][player]["spaceship"]["angle"] = 0;
+    serverData["sessions"][session]["players"][player]["spaceship"]["resources"] = 0;
+    serverData["sessions"][session]["players"][player]["planet"]["position"] = [50, 50];
+    serverData["sessions"][session]["players"][player]["planet"]["resources"] = 0;
+    serverData["sessions"][session]["players"][player]["spaceship"]["isSpectating"] = false;
+  });
 
-  sendMessageToSession( "sessionReset", { "info": serverData[ "sessions" ][ session ] }, session );
+  sendMessageToSession("sessionReset", { "info": serverData["sessions"][session] }, session);
 }
 
-function resetSessionTime( session ) {
-  serverData[ "sessions" ][ session ][ "remainingTime" ] = serverData[ "sessions" ][ session ][ "time" ];
+function resetSessionTime(session) {
+  serverData["sessions"][session]["remainingTime"] = serverData["sessions"][session]["time"];
 }
 
 function updateTime() {
-  Object.keys( serverData[ "sessions" ] ).forEach( ( session ) => {
-    if ( serverData[ "sessions" ][ session ][ "state" ] == "playing" ) {
-      serverData[ "sessions" ][ session ][ "remainingTime" ] -= 1000;
-
-      sendMessageToSession( "timeUpdated", { "remainingTime": serverData[ "sessions" ][ session ][ "remainingTime" ] }, session );
-
-      updateSessionState( session )
+  Object.keys(serverData["sessions"]).forEach((session) => {
+    if (serverData["sessions"][session]["state"] == "playing") {
+      serverData["sessions"][session]["remainingTime"] -= 1000;
```

Comment:
I should change that to seconds instead of milliseconds, like you're doing with the respawn timer.